### ISSUE TITLE
Issue paging is pointless because of randomization

### ISSUE THREAD

**Issue by Mr0grog** [codeforamerica/cfapi#167]
There might be something obvious I'm missing here, but it seems like pagination for the issues API (`/api/issues` and `/api/issues/labels/<labels>`) is completely pointless and broken because it intentionally randomizes the results.

For example, I made consecutive requests to:
`http://codeforamerica.org/api/issues/labels/help%20wanted?per_page=100&page=1`
`http://codeforamerica.org/api/issues/labels/help%20wanted?per_page=100&page=2`

And wound up with 26 issues that appeared in both responses, meaning I never actually got all 143 results that I should have (the value of `total` in both responses was `143`). Page N is never really reliably page N because the set of results is randomized on every request.

I'm guessing this was done on purpose (maybe for Civic Tech Issue Finder?), but it seems like the need would be better served by having an `order` parameter that could be set to `random` or a specific `randomized=true` parameter or something similar. As it stands, it's impossible to use this API to actually fully iterate through any set of issues.

**Comment by ondrae** [comment]
You're right. We've made issues [return randomly](https://github.com/codeforamerica/cfapi/blob/master/app.py#L780) so that the Issue Finder was always fresh, which defeats the point of paging for Issues.

We should add some `order` params.

My quick take of it would be something like:
```
    query = db.session.query(Issue)

    for attr, value in filters.iteritems():
        if 'order' in attr:
             if value == 'newest':
                query = query.order_by(desc(Issue.id))
             if value == 'oldest':
                query = query.order_by(asc(Issue.id))
             if value == 'random':
                query = query.order_by(func.random())
```

**Comment by Mr0grog** [comment]
Quick thoughts:

1. Are issue IDs unique and increasing across projects? Will issues always
only be Github issues? (plenty of reasons for projects to use other
trackers.) If either of those are a no, then I'm thinking ID probably isn't
a great field for sorting. Maybe date created or updated?

2. Traditionally, API sorting distinguishes between the field you're
sorting on (usually `order_by`) and the sort order (usually `order` valued
as ascending/descending). Do you think the API will ever need to support
sorting on other fields? If so, ascending/descending instead of
newest/oldest are probably better terms. Or a totally separate field to
specify random.

3. If no order or sorting is specified, there should still be defaults—that
way you can guarantee stability across pages.
